common/result_test.go

```diff
@@ -0,0 +1,101 @@
+package common
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLabel(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	intLabel := IntLabel(1)
+	assert.True(intLabel.IsInt())
+	assert.Equal("1", intLabel.String())
+
+	bytes, err := json.Marshal(intLabel)
+	require.Nil(err)
+	assert.Equal("\"1\"", string(bytes))
+
+	stringLabel := StringLabel("test")
+	assert.False(stringLabel.IsInt())
+	assert.Equal("1", intLabel.String())
+
+	bytes, err = json.Marshal(stringLabel)
+	require.Nil(err)
+	assert.Equal("\"test\"", string(bytes))
+
+}
+
+func TestResult_RoundTrip(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	result := AttestationResult{
+		Status: StatusSuccess,
+		TrustVector: TrustVector{
+			HardwareAuthenticity: StatusSuccess,
+			SoftwareIntegrity:    StatusSuccess,
+			SoftwareUpToDateness: StatusSuccess,
+			ConfigIntegrity:      StatusSuccess,
+			RuntimeIntegrity:     StatusSuccess,
+			CertificationStatus:  StatusUnknown,
+		},
+		RawEvidence:       []byte{0xDE, 0xAD, 0xBE, 0xEF},
+		Timestamp:         time.Date(1970, 1, 1, 0, 0, 0, 0, time.UTC),
+		EndorsedClaims:    EndorsedClaims{},
+		AppraisalPolicyID: "test-policy",
+	}
+
+	bytes, err := result.ToJSON()
+	require.Nil(err)
+	assert.Equal(`{"veraison-processed-evidence":{},"status":1,"trust-vector":{"hw-authenticity":1,"sw-integrity":1,"sw-up-to-dateness":1,"config-integrity":1,"runtime-integrity":1,"certification-status":2},"raw-evidence":"3q2+7w==","timestamp":"1970-01-01T00:00:00Z","endorsed-claims":{"hw-details":{},"sw-details":{},"certification-details":{},"config-details":{}},"appraisal-policy-id":"test-policy"}`, string(bytes))
```

Comment:
(note) you could use [JSONEq](https://pkg.go.dev/github.com/stretchr/testify/assert#JSONEq)